v2/originating_identity.go

```diff
@@ -0,0 +1,87 @@
+package v2
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+func buildOriginatingIdentityHeaderValue(i OriginatingIdentity) (string, error) {
+	if i == nil {
+		return "", nil
+	}
+	platform := i.Platform()
+	value, err := i.Value()
+	if err != nil {
+		return "", err
+	}
+	encodedValue := base64.StdEncoding.EncodeToString([]byte(value))
+	headerValue := fmt.Sprintf("%v %v", platform, encodedValue)
+	return headerValue, nil
+}
+
+var _ OriginatingIdentity = &CloudFoundryOriginatingIdentity{}
+
+func (i *CloudFoundryOriginatingIdentity) Platform() string {
+	return "cloudfoundry"
+}
+
+func (i *CloudFoundryOriginatingIdentity) Value() (string, error) {
+	if err := i.validate(); err != nil {
+		return "", err
+	}
+	baseProperties := map[string]interface{}{
+		"user_id": i.UserId,
+	}
+	return marshalValue(baseProperties, i.Extra)
+}
+
+func (i *CloudFoundryOriginatingIdentity) validate() error {
+	if len(i.UserId) == 0 {
+		return errors.New("UserId is required")
+	}
+	return nil
+}
+
+var _ OriginatingIdentity = &KubernetesOriginatingIdentity{}
+
+func (i *KubernetesOriginatingIdentity) Platform() string {
+	return "kubernetes"
+}
+
+func (i *KubernetesOriginatingIdentity) Value() (string, error) {
+	if err := i.validate(); err != nil {
+		return "", err
+	}
+	baseProperties := map[string]interface{}{
+		"username": i.Username,
+		"uid":      i.Uid,
```

Comment:
We need to send more than just the username and UID; need to also send groups and extrainfo.